src/plugins/data/public/query/query_string/query_history.ts

```diff
@@ -14,12 +14,12 @@ export const HISTORY_KEY_PREFIX = 'query_';
 export class QueryHistory {
   private changeEmitter: BehaviorSubject<any[]>;
 
-  constructor(private readonly storage: DataStorage) {
+  constructor(private readonly sessionStorage: DataStorage) {
```

Comment:
nit: do you mind restoring this? i think it might be a little bit confusing.

i might have been unclear. `DataStorage` is the the storage class that has an storage engine of either `sessionStorage` or `localStorage`.

I think calling this `sessionStorage` might be confusing and potentially incorrect because I can swap the engine under dataStorage without it being transparent. `storage` is more generic.

Reply:
as you can see https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/data/public/plugin.ts#L132 i could modify this to be window.localStorage which could leave us open for regressions later.

Reply:
Yes `DataStorage` is a general storage class, one of its function `createStorage` can be used to create either a localStorage or sessionStorage. https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/data/public/plugin.ts#L130

We should change this to storage or dataStorage perhaps?

Reply:
yeah i can see the case where a regression can happen that is obfuscated with the naming since someone can change the the engine used to be `window.localStorage` but then it gets passed down into query_history as renamed to `sessionStorage` when it is actually using localStorage

Reply:
@abbyhu2000 @kavilla thanks for reviewing. I added a follow-up PR https://github.com/opensearch-project/OpenSearch-Dashboards/pull/8733